app/web/features/auth/phone/ChangePhone.tsx

```diff
@@ -0,0 +1,258 @@
+import "react-phone-number-input/style.css";
+
+import { Typography, useMediaQuery, useTheme } from "@material-ui/core";
+import Alert from "components/Alert";
+import Button from "components/Button";
+import TextField from "components/TextField";
+import { accountInfoQueryKey } from "features/queryKeys";
+import { Empty } from "google-protobuf/google/protobuf/empty_pb";
+import { RpcError } from "grpc-web";
+import luhn from "luhn";
+import { Trans, useTranslation } from "next-i18next";
+import { GetAccountInfoRes } from "proto/account_pb";
+import { Controller, useForm } from "react-hook-form";
+import PhoneInput, {
+  formatPhoneNumberIntl,
+  isValidPhoneNumber,
+} from "react-phone-number-input";
+import { useMutation, useQueryClient } from "react-query";
+import { service } from "service";
+
+import useChangeDetailsFormStyles from "../useChangeDetailsFormStyles";
+
+export const validatePhoneCode = (code: string) =>
+  code.length == 6 && luhn.validate(code);
+
+interface ChangePhoneFormData {
+  phone: string;
+}
+
+interface VerifyPhoneFormData {
+  code: string;
+}
+
+type ChangePhoneProps = GetAccountInfoRes.AsObject & {
+  className?: string;
+};
```

Comment:
```suggestion
interface ChangePhoneProps {
  accountInfo: GetAccountInfoRes.AsObject;
  className?: string;
};
```
To promote the prop being passed into the component explicitly